fgpyo/io/__init__.py

```diff
@@ -124,30 +128,32 @@ def assert_path_is_writable(path: Path, parent_must_exist: bool = True) -> None:
     Example:
         assert_path_is_writable(path = Path("example.txt"))
     """
-    # If file exists, it must be writable
+    # stdout is writable
+    if path == Path("/dev/stdout"):
+        return
+
+    # If path exists, it must be a writable file
     if path.exists():
-        assert path.is_file() and os.access(
-            path, os.W_OK
-        ), f"File exists but is not writable: {path}"
+        assert path.is_file(), f"Cannot read path becasue it is not a file: {path}"
+        assert os.access(path, os.W_OK), f"File exists but is not writable: {path}"
 
     # Else if file doesnt exist and parent_must_exist is True then check
     # that path.absolute().parent exists, is a directory and is writable
     elif parent_must_exist:
         parent = path.absolute().parent
-        assert (
-            parent.exists() & parent.is_dir() & os.access(parent, os.W_OK)
-        ), f"Path does not exist and parent isn't extant/writable: {path}"
+        assert parent.exists(), f"Parent directory does not exist: {parent}"
+        assert parent.is_dir(), f"Parent directory exists but is not a directory: {parent}"
+        assert os.access(parent, os.W_OK), f"Parent directory exists but is not writable: {parent}"
 
     # Else if file doesn't exist and parent_must_exist is False, test parent until
     # you find the first extant path, and check that it is a directory and is writable.
     else:
-        for parent in path.parents:
+        for parent in path.absolute().parents:
```

Comment:
@nh13 @tfenne it wasn't clear to me why we only considered the absolute path when `parent_must_exist=True`. If the omission was intentional please lmk and I'll revert

Reply:
Taking a shit in the dark, what if the path includes a symbolic link?

Reply:
Does it matter? Won't the symbolic link inherit the permissions of the target?

Reply:
Resolved per slack discussion

> `Path.is_file()` returns `True` for symlinks


> Let's say you have the following
> ```
> /heres/the/symlinked/path/to/file -> /and/the/actual/path/to/file
> ```
> In the context of the assert, the question is basically only relevant when `file`, `to`, and `path` do not exist
> and afaik a symlinked directory is writable iff the target directory is writable